### ISSUE TITLE
feat: btop-inspired GPU monitoring for terminal training dashboard

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#74]
## Summary

Add real-time GPU monitoring to the terminal training dashboard (`src/train/tui.rs`), inspired by [btop](https://github.com/aristocratos/btop)'s GPU visualization. This will provide live hardware utilization metrics during neural network training.

## Motivation

Training neural networks involves significant GPU utilization. Currently, `TerminalMonitorCallback` tracks loss, learning rate, and training progress, but lacks hardware monitoring. Users must run separate tools (nvidia-smi, btop) to monitor GPU health during training.

A unified dashboard showing both training metrics AND hardware utilization enables:
- Early detection of GPU thermal throttling
- Memory pressure identification (OOM prevention)
- Power efficiency monitoring
- PCIe bottleneck detection

### 1. GPU Metrics Collection (via NVML)

Reference: btop's `src/linux/btop_collect.cpp:106-157` (Nvml namespace)

```rust
/// GPU metrics snapshot
pub struct GpuMetrics {
    pub name: String,
    pub utilization_percent: u32,      // GPU compute utilization
    pub memory_used_mb: u64,
    pub memory_total_mb: u64,
    pub memory_utilization_percent: u32,
    pub temperature_celsius: u32,
    pub power_watts: f32,
    pub power_limit_watts: f32,
    pub clock_mhz: u32,
    pub memory_clock_mhz: u32,
    pub pcie_tx_kbps: u64,
    pub pcie_rx_kbps: u64,
}
```

### 2. NVML Integration

Use dynamic loading like btop (`dlopen`/`dlsym`) for optional GPU support:
- `nvmlInit()` / `nvmlShutdown()`
- `nvmlDeviceGetUtilizationRates()`
- `nvmlDeviceGetMemoryInfo()`
- `nvmlDeviceGetTemperature()`
- `nvmlDeviceGetPowerUsage()`
- `nvmlDeviceGetClockInfo()`
- `nvmlDeviceGetPcieThroughput()`

Consider using existing Rust bindings: `nvml-wrapper` crate.

### 3. Dashboard Layout Extensions

Reference: btop's `src/btop_draw.cpp` and `src/btop_shared.hpp:130-171`

**New `DashboardLayout::Full` with GPU panel:**

```
═══ llama-7b Training ═══
Epoch 3/10 │ loss=0.0234 val=0.0256 best=0.0198
Loss: ▁▂▃▄▅▆▅▄▃▂▁▂▃▂▁▂▃▄▃▂ 0.0234
LR:   ▇▇▆▆▅▅▄▄▃▃▃▂▂▂▂▁▁▁▁▁ 1.00e-04
──────────────────────────────────────────
GPU 0: RTX 4090 │ 87% │ 45°C │ 285W/450W
VRAM: ████████████████░░░░ 18.2/24.0 GB (76%)
Mem:  ▁▂▃▄▅▆▇█▇▆▅▄▃▂▁▂▃▄▅▆
──────────────────────────────────────────
[████████████████████░░░░░░░░░░] 67% ETA: 12m
```

### 4. Multi-GPU Support

Track metrics per device, aggregate for `DashboardLayout::Compact`:
- Show primary GPU in compact view
- Full GPU list in `DashboardLayout::Full`

### 5. Alert Integration (AndonSystem)

Extend `AndonSystem` with GPU alerts:
- `GpuThermalThrottling`: temp > 85°C
- `GpuMemoryPressure`: VRAM > 95%
- `GpuPowerLimit`: power > 95% of limit

## Files to Modify

- `src/train/tui.rs` - Add GPU metrics display
- `src/train/mod.rs` - Export new types
- New: `src/gpu/mod.rs` - GPU metrics collection
- New: `src/gpu/nvml.rs` - NVML bindings/wrapper

## References

- btop GPU collection: `src/linux/btop_collect.cpp` lines 106-211
- btop GPU struct: `src/btop_shared.hpp` lines 130-171
- btop GPU drawing: `src/btop_draw.cpp`
- NVML API: https://docs.nvidia.com/deploy/nvml-api/

## Acceptance Criteria

- [ ] GPU utilization, temperature, power, VRAM displayed in terminal
- [ ] Sparkline history for GPU metrics (reuse `MetricsBuffer`)
- [ ] Multi-GPU support (enumerate all devices)
- [ ] Graceful degradation when NVML unavailable
- [ ] AndonSystem alerts for thermal/memory issues
- [ ] < 10ms overhead per refresh cycle
- [ ] Unit tests with mock NVML responses
- [ ] Documentation with screenshots

## Dependencies

- Optional: `nvml-wrapper` crate or raw FFI bindings
- Feature flag: `gpu-monitoring` (default enabled on Linux)

**Comment by noahgift** [comment]
Implemented in src/monitor/gpu/ with GPU monitor (monitor.rs), alert system (alert/), ring buffer (buffer.rs), terminal rendering (render.rs), and type definitions. Provides btop-inspired GPU monitoring for training dashboards.
